conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,700 @@
+    error err6 = error E("message 6", err1);
+    error err7 = error E("message 7", fatal = false);
+    error err8 = error E("message 8", err1, fatal = false);
+
+    io:println(err1.message()); // @output message 1
+    io:println(err2.message()); // @output message 2
+    io:println(err3.message()); // @output message 3
+    io:println(err4.message()); // @output message 4
+    io:println(err5.message()); // @output message 5
+    io:println(err6.message()); // @output message 6
+    io:println(err7.message()); // @output message 7
+    io:println(err8.message()); // @output message 8
+}
+
+Test-Case: output
+Description: Test error constructor constructing a new error value.
+Labels: anydata, error-constructor-expr, error-type, error-type-parameter, is-expr, map-type, module-type-defn, string
+
+type E error<map<anydata>>;
+
+function init() {
+    io:println(error("error message") is error); // @output true
+    io:println(error("error message", msg = "message", fatal = false) is error); // @output true
+    io:println(error("error message", error("ERROR!")) is error); // @output true
+    io:println(error("error message", error("ERROR!", errorCode = "BCE0001"), msg = "message", fatal = false) is error); // @output true
+
+    io:println(error E("error message") is error); // @output true
+    io:println(error E("error message", msg = "message", fatal = false) is error); // @output true
+    io:println(error E("error message", error("ERROR!")) is error); // @output true
+    io:println(error E("error message", error E("ERROR!", errorCode = "BCE0001"), msg = "message", fatal = false) is error); // @output true
+}
+
+Test-Case: error
+Description: Test specified error-type-reference has to be a subtype of error.
+Labels: any, error-constructor-expr, error-type, module-type-defn, string, union-type, var
+
+type E1 any;
+
+type E2 any|error;
+
+function errorFunction() {
+    error err1 = error E1("error message"); // @error cannot create a new error value from 'E1'
+    error err2 = error E2("error message"); // @error cannot create a new error value from 'E2'
+    var err3 = error E1("error message"); // @error cannot create a new error value from 'E1'
+    var err4 = error E2("error message"); // @error cannot create a new error value from 'E2'
+}
+
+Test-Case: error
+Description: Test the effect of specified error-type-reference is same as making the contextually expected type be that specified type reference.
+Labels: error-constructor-expr, error-type, error-type-parameter, module-type-defn, record-type, string, var
+
+type E error<record {|string message;|}>;
+
+function errorFunction() {
+    E err1 = error("error message"); // @error missing error detail arg for error detail field 'message'
+
+    var err2 = error E("error message"); // @error missing error detail arg for error detail field 'message'
+
+    error err3 = error E("error message"); // @error missing error detail arg for error detail field 'message'
```

Comment:
Maybe a case with intersection-type-desc as CEP

Reply:
There is a separate test with CEP being intersection type at L488 if that's what you meant.